dynamodb-enhanced/deployment/src/main/java/io/quarkus/amazon/dynamodb/enhanced/runtime/EnhancedProcessor.java

```diff
@@ -0,0 +1,200 @@
+package io.quarkus.amazon.dynamodb.enhanced.runtime;
+
+import java.util.function.BiFunction;
+import java.util.function.Consumer;
+
+import org.jboss.jandex.AnnotationInstance;
+import org.jboss.jandex.ClassInfo;
+import org.jboss.jandex.DotName;
+import org.objectweb.asm.AnnotationVisitor;
+import org.objectweb.asm.ClassVisitor;
+import org.objectweb.asm.MethodVisitor;
+import org.objectweb.asm.Opcodes;
+import org.objectweb.asm.Type;
+
+import io.quarkus.arc.deployment.AdditionalBeanBuildItem;
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.builditem.BytecodeTransformerBuildItem;
+import io.quarkus.deployment.builditem.CombinedIndexBuildItem;
+import io.quarkus.deployment.builditem.FeatureBuildItem;
+import io.quarkus.deployment.builditem.LaunchModeBuildItem;
+import io.quarkus.deployment.builditem.nativeimage.ReflectiveClassBuildItem;
+import io.quarkus.gizmo.Gizmo;
+import io.quarkus.runtime.LaunchMode;
+import software.amazon.awssdk.enhanced.dynamodb.mapper.annotations.DynamoDbBean;
+
+public class EnhancedProcessor {
+    private static final String FEATURE = "amazon-dynamodb-enhanced";
+    public static final String CLASS_NAME_BEAN_TABLE_SCHEMA = "software.amazon.awssdk.enhanced.dynamodb.mapper.BeanTableSchema";
+
+    @BuildStep
+    FeatureBuildItem feature() {
+        return new FeatureBuildItem(FEATURE);
+    }
+
+    @BuildStep
+    AdditionalBeanBuildItem producer() {
+        return AdditionalBeanBuildItem.unremovableOf(DynamodbEnhancedClientProducer.class);
+    }
+
+    @BuildStep
+    public void registerClassesForReflectiveAccess(
+            CombinedIndexBuildItem combinedIndexBuildItem,
+            BuildProducer<ReflectiveClassBuildItem> reflectiveClass) {
+        for (AnnotationInstance i : combinedIndexBuildItem
+                .getIndex()
+                .getAnnotations(DotName.createSimple(DynamoDbBean.class.getName()))) {
+            ClassInfo classInfo = i.target().asClass();
+            reflectiveClass.produce(
+                    new ReflectiveClassBuildItem(true, false, classInfo.name().toString()));
+        }
+
+        // When we add DefaultAttributeConverterProvider and BeanTableSchemaAttributeTags here, we do not need the DynamodbEnhancedFeature in runtime module.
+        //reflectiveClass.produce(
+        //        new ReflectiveClassBuildItem(true, false,
+        //                "software.amazon.awssdk.enhanced.dynamodb.DefaultAttributeConverterProvider"));
+        //reflectiveClass.produce(
+        //        new ReflectiveClassBuildItem(true, false,
+        //                "software.amazon.awssdk.enhanced.dynamodb.internal.mapper.BeanTableSchemaAttributeTags"));
+
```

Comment:
That's not surprising, see comment above about where you should place the dependency :).